src/security-council-mgmt/SecurityCouncilUpgradeAction.sol

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.16;
+
+import "./interfaces/IGnosisSafe.sol";
+import "./SecurityCouncilMgmtUtils.sol";
+
+/// @notice Action contract for updating security council members. Used by the security council management system.
+contract SecurityCouncilUpgradeAction {
+    address internal constant SENTINEL_OWNERS = address(0x1);
+
+    IGnosisSafe public immutable securityCouncil;
+
+    constructor(address _securityCouncil) {
+        securityCouncil = IGnosisSafe(_securityCouncil);
+    }
+
+    /// @notice updates members of security council multisig to match provided array
+    function updateMembers(address[] memory _updatedMembers) external {
+        // always preserve current threshold
+        uint256 threshold = securityCouncil.getThreshold();
+
+        address[] memory membersToAdd = new address[](6);
+        uint8 membersToAddCount = 0;
+
+        for (uint256 i = 0; i < _updatedMembers.length; i++) {
+            address member = _updatedMembers[i];
+            if (!securityCouncil.isOwner(member)) {
+                membersToAdd[membersToAddCount] = member;
+                membersToAddCount++;
+            }
+        }
+
+        address[] memory membersToRemove = new address[](6);
+        uint8 membersToRemoveCount = 0;
+
+        address[] memory owners = securityCouncil.getOwners();
+        for (uint256 i = 0; i < owners.length; i++) {
+            address owner = owners[i];
+            if (!SecurityCouncilMgmtUtils.isInArray(owner, _updatedMembers)) {
+                membersToRemove[membersToRemoveCount] = owner;
+                membersToRemoveCount++;
+            }
+        }
+
+        for (uint256 i = 0; i < membersToAddCount; i++) {
+            _addMember(membersToAdd[i], threshold);
+        }
+        for (uint256 i = 0; i < membersToRemoveCount; i++) {
+            _removeMember(membersToRemove[i], threshold);
+        }
```

Comment:
why don’t we use `swapOwner` instead of `addOwnerWithThreshold` and `removeOwner`?
https://github.com/safe-global/safe-contracts/blob/64a0851ad31a46d0848f9ee1ca30a560d2d42f1e/contracts/base/OwnerManager.sol#L92-L99

Reply:
i played with it but it was too weird and edge-casey. e.g., you would need a different path for elections (all swaps), vs. addMember, and removeMember. Also for the election case, figuring out what to swap with what gets strange and complex (since there's shared members in old and new, and each swap modified the linked list, etc)

Reply:
open to suggestions tho